x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRuleCriteria.java

```diff
@@ -167,20 +192,57 @@ public boolean equals(Object o) {
         QueryRuleCriteria that = (QueryRuleCriteria) o;
         return criteriaType == that.criteriaType
             && Objects.equals(criteriaMetadata, that.criteriaMetadata)
-            && Objects.equals(criteriaValue, that.criteriaValue);
+            && Objects.equals(criteriaValues, that.criteriaValues);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(criteriaType, criteriaMetadata, criteriaValue);
+        return Objects.hash(criteriaType, criteriaMetadata, criteriaValues);
     }
 
     @Override
     public String toString() {
         return Strings.toString(this);
     }
 
-    public boolean isMatch(String matchString) {
-        return criteriaType == CriteriaType.EXACT && criteriaValue.equals(matchString);
+    public boolean isMatch(Object matchValue, CriteriaType matchType) {
+        if (criteriaType == GLOBAL) {
+            return true;
+        }
+
+        final LevenshteinDistance ld = new LevenshteinDistance();
+        final String matchString = matchValue.toString();
+        boolean matchFound = false;
+
+        for (Object criteriaValue : criteriaValues) {
+            final String criteriaValueString = criteriaValue != null ? criteriaValue.toString() : null;
+            if (criteriaValueString != null && matchValue instanceof String) {
+                matchFound |= switch (matchType) {
+                    case EXACT -> matchString.equals(criteriaValueString);
+                    case EXACT_FUZZY -> ld.getDistance(matchString, criteriaValueString) > 0.75f; // TODO make configurable
```

Comment:
What's a good default value for edit distance here? We have a followup [issue](https://github.com/elastic/enterprise-search-team/issues/5400) to make this configurable as it involves an API change, so I'd like to start with a good default value.

Reply:
Something familiar to users could be the `fuziness` parameter in ES:

https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#fuzziness

This defines the max number of edits, and also uses `AUTO` to match based on the lenght of the word.

Reply:
Maybe we should remove the fuzzy matching until we can make it configurable or have a more complete feature?

Reply:
I'd like to get it into the release, but if it doesn't make it in perhaps we could decide not to highlight it in the docs?

Reply:
I'm not sure how much useful it is as it stands now, and I think it can be added later with little effort. Not a blocker, it's just that I think we can do better.

Reply:
Agreed, it's slated for a followup PR!